### PR TITLE
Fixed fullscreen and orientation handling to work with SDL-2.0.9 on Android

### PR DESCRIPTION
**Pull request by inclement**
Makes Kivy compatible with python-for-android after the changes in https://github.com/kivy/python-for-android/pull/1528.

This probably also fixes some bugs in reading the env vars in python2/python3, especially on iOS with python3.

The code here was a bit messy and could do with further improvement, but that's a more general Kivy issue.

There is still some problem with handling the back button. After pressing it so that the app closes, then re-opening the app, I get the following error:

```
01-20 16:49:16.435 18071 18160 I python  : [INFO   ] [Logger      ] Record log in /data/user/0/net.inclem.lzviewer/files/app/.kivy/logs/kivy_19-01-20_1.txt
01-20 16:49:16.435 18071 18160 I python  : [INFO   ] [Kivy        ] v1.11.0.dev0, git-12a8b13, 20190120
01-20 16:49:16.435 18071 18160 I python  : [INFO   ] [Python      ] v3.7.1 (default, Jan 18 2019, 23:21:04) 
01-20 16:49:16.435 18071 18160 I python  : [Clang 6.0.2 (https://android.googlesource.com/toolchain/clang 183abd29fc496f55
01-20 16:49:16.669 18071 18160 I python  :  Traceback (most recent call last):
01-20 16:49:16.669 18071 18160 I python  :    File "main.py", line 14, in <module>
01-20 16:49:16.670 18071 18160 I python  :      from kivy.app import App
01-20 16:49:16.670 18071 18160 I python  :    File "/data/user/0/net.inclem.lzviewer/files/app/_python_bundle/site-packages/kivy/app.py", line 319, in <module>
01-20 16:49:16.671 18071 18160 I python  :      from kivy.base import runTouchApp, stopTouchApp
01-20 16:49:16.672 18071 18160 I python  :    File "/data/user/0/net.inclem.lzviewer/files/app/_python_bundle/site-packages/kivy/base.py", line 26, in <module>
01-20 16:49:16.672 18071 18160 I python  :      from kivy.clock import Clock
01-20 16:49:16.672 18071 18160 I python  :    File "/data/user/0/net.inclem.lzviewer/files/app/_python_bundle/site-packages/kivy/clock.py", line 706, in <module>
01-20 16:49:16.674 18071 18160 I python  :      class ClockBase(ClockBaseBehavior, CyClockBase):
01-20 16:49:16.675 18071 18160 I python  :  TypeError: metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all its bases
01-20 16:49:16.861 18071 18160 I python  : Python for android ended.
```
I haven't yet checked if this is related to the SDL_ANDROID_TRAP_BACK_BUTTON hint, which I don't think I have set. It's a weird error though, the Python thread appears to finish and close just fine so I don't see what state is hanging around to cause this problem. The issue goes away if the app is completely closed by swiping it away in the app list.

### CODE REVIEW

**Comment by inclement** [comment]
If I set the SDL_ANDROID_TRAP_BACK_BUTTON hint then I receive the back button as a keypress, as expected, and if I tell Kivy to ignore it by the normal mechanism then the back button is ignored as expected. However, if I ignore it and let Kivy close then the metaclass issue always occurs until I manually force close the app. I'm a bit perplexed at this one, @JonasT have you seen anything like this?

Kivy appears to be closing fine, the same as ever:
```
01-20 17:41:32.449 24251 24268 I python  : [INFO   ] [Base        ] Leaving application in progress...
01-20 17:41:32.468 24251 24268 I python  : [INFO   ] [WindowSDL   ] exiting mainloop and closing.

```

**Comment by JonasT** [comment]
@inclement no clue, I pretty much added the back button hint instantly to my non-kivy app and I don't think I've ever seen such a metaclass crash ever :woman_shrugging:

**Comment by inclement** [comment]
I tried another test app and that one seems to work, which makes me suspect the issue is due to my more complicated app not cleaning up a thread properly. Still investigating, but it hopefully this will reveal that Kivy basically works fine.

**Comment by inclement** [comment]
@AndreMiras I updated this to handle the back button correctly, could you review it again?

This doesn't fix the issue where the app crashes with a traceback about metaclasses if it's reopened after closing with the back button. I can't work out why this happens, but it can be 'fixed' with `sys.exit(0)`, even if that is placed right at the end of the existing app after `App.run()` has completed and returned. I'm not sure what to do about that, but at least understanding it this much this PR can be merged safely.

File: kivy/core/window/_window_sdl2.pyx [hunk 1]
```diff
@@ -89,28 +95,39 @@ cdef class _WindowSDL2Storage:
 
         SDL_SetHint(SDL_HINT_ACCELEROMETER_AS_JOYSTICK, b'0')
 
+        SDL_SetHintWithPriority(b'SDL_ANDROID_TRAP_BACK_BUTTON', b'1',
```
Comment (inclement) [inline kivy/core/window/_window_sdl2.pyx]: As far as I can tell from the code, SDL2 doesn't care if we set this even in older versions where the hint doesn't exist. I didn't explicitly test this though.

**Pull request merged**